cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/asset_transfers.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+
+fn send_asset_from_asset_hub_rococo_to_asset_hub_westend(id: MultiLocation, amount: u128) {
+	let signed_origin =
+		<AssetHubRococo as Chain>::RuntimeOrigin::signed(AssetHubRococoSender::get().into());
+	let asset_hub_westend_para_id = AssetHubWestend::para_id().into();
+	let destination = MultiLocation {
+		parents: 2,
+		interior: X2(GlobalConsensus(NetworkId::Westend), Parachain(asset_hub_westend_para_id)),
+	};
+	let beneficiary_id = AssetHubWestendReceiver::get();
+	let beneficiary: MultiLocation =
+		AccountId32Junction { network: None, id: beneficiary_id.into() }.into();
+	let assets: MultiAssets = (id, amount).into();
+	let fee_asset_item = 0;
+
+	// fund the AHR's SA on BHR for paying bridge transport fees
+	let ahr_as_seen_by_bhr = BridgeHubRococo::sibling_location_of(AssetHubRococo::para_id());
+	let sov_ahr_on_bhr = BridgeHubRococo::sovereign_account_id_of(ahr_as_seen_by_bhr);
+	BridgeHubRococo::fund_accounts(vec![(sov_ahr_on_bhr.into(), 10_000_000_000_000u128)]);
+
+	AssetHubRococo::execute_with(|| {
+		assert_ok!(
+			<AssetHubRococo as AssetHubRococoPallet>::PolkadotXcm::limited_reserve_transfer_assets(
+				signed_origin,
+				bx!(destination.into()),
+				bx!(beneficiary.into()),
+				bx!(assets.into()),
+				fee_asset_item,
+				WeightLimit::Unlimited,
+			)
+		);
```

Comment:
Maybe also check that we have withdrawn assets from origin account and deposited it to the local dest account? I guess there'll be some events on `AssetHubRococo` for that?

Reply:
I see there are checks like `assert!(sender_rocs_before > sender_rocs_after);` in the caller functions - so maybe instead of events also check that `reserve_account_rocs_before < reserve_account_rocs_after` if checking events is hard here?